### ISSUE TITLE
strcred.py is missing from the Python3 installation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8216]
|<img alt="doko's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| doko reported|
|-|-|
|Trac ID|trac#8216|
|Type|release blocker: regression|
|Created|2016-02-24 05:38:43Z|
|Branch|https://github.com/twisted/twisted/tree/strcred-dist3-8216|

[reported at https://bugs.launchpad.net/bugs/1548972]

seen with 16.0.0pre1, calling the twistd script with Python3:

```
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/twisted/python/usage.py", line 447, in __str__
    return self.getSynopsis() + '\n' + self.getUsage(width=None)
  File "/usr/lib/python3/dist-packages/twisted/python/usage.py", line 483, in getUsage
    for (cmd, short, parser, desc) in self.subCommands:
  File "/usr/lib/python3/dist-packages/twisted/application/app.py", line 596, in subCommands
    for plug in sorted(plugins, key=attrgetter('tapname')):
  File "/usr/lib/python3/dist-packages/twisted/plugin.py", line 213, in getPlugins
    allDropins = getCache(package)
--- <exception caught here> ---
  File "/usr/lib/python3/dist-packages/twisted/plugin.py", line 171, in getCache
    provider = pluginModule.load()
  File "/usr/lib/python3/dist-packages/twisted/python/modules.py", line 389, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/usr/lib/python3/dist-packages/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/usr/lib/python3/dist-packages/twisted/python/reflect.py", line 250, in _importAndCheckStack
    reraise(excValue, excTraceback)
  File "/usr/lib/python3/dist-packages/twisted/python/compat.py", line 412, in reraise
    raise exception.with_traceback(traceback)
  File "/usr/lib/python3/dist-packages/twisted/plugins/cred_sshkeys.py", line 15, in <module>
    from twisted.cred.strcred import ICheckerFactory
builtins.ImportError: No module named 'twisted.cred.strcred'
```

<details><summary>Searchable metadata</summary>

```
trac-id__8216 8216
type__release_blocker__regression release blocker: regression
reporter__doko doko
priority__highest highest
milestone__Twisted_16_0 Twisted 16.0
branch__branches_strcred_dist3_8216 branches/strcred-dist3-8216
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1456292323936144 1456292323936144
changetime__1456629347067288 1456629347067288
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_2"></a>|@mithrandi commented|
|-|-|

Looks like the `cred_sshkeys` plugin was added to `dist3` in [#7998](https://github.com/twisted/twisted/issues/7998), but `twisted.cred.strcred` was not. I think this was an oversight?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

Yeah, this is an oversight, and a release blocker :((((((((((((((((((((((((

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

(In [46838]) Branching to strcred-dist3-8216.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

strcred wasn't actually ported, so we can't add it -- so, the easiest thing was to just port it.

Builders spun, all green. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_6"></a>|@mithrandi set owner to @hawkowl|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_7"></a>|@mithrandi set status to `closed`|
|-|-|

(In [46842]) Merge strcred-dist3-8216: port twisted.cred.strcred to Python 3.

Fixes: [#8216](https://github.com/twisted/twisted/issues/8216)
Author: hawkowl
Reviewer: mithrandi

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

(In [46860]) Merge strcred-dist3-8216: port twisted.cred.strcred to Python 3.

Fixes: [#8216](https://github.com/twisted/twisted/issues/8216)
Author: hawkowl
Reviewer: mithrandi
